src/components/Activity/NewRequestsRow.js

```diff
@@ -36,22 +40,28 @@ class NewRequestsRow extends Component {
     } else if (result) {
       MatchUtil.getActivityList(this.onGetActivityCallback.bind(this));
     }
+
+    this.setState({ requestPending: false });
   }
 
   acceptRequest() {
-    if (this.props.dataSource.close) {
-      MatchUtil.acceptRequest(this.onRequestCallback.bind(this), this.props.dataSource._id);
-    } else {
-
+    if (this.state.isSwipeOpened) {
       // Call this method with no parameter will close all swipe button
       this.props.closeAllExceptCurrent();
+    } else {
+      MatchUtil.acceptRequest(this.onRequestCallback.bind(this), this.props.dataSource._id);
+      this.setState({ requestPending: true });
```

Comment:
This boolean variable managing state to check whether user's network request is in progress or not